pkg/extensions/search/schema.resolvers.go

```diff
@@ -494,6 +495,87 @@ func (r *queryResolver) GlobalSearch(ctx context.Context, query string) (*gql_ge
 	}, nil
 }
 
+// BaseImageList is the resolver for the BaseImageList field.
+func (r *queryResolver) BaseImageList(ctx context.Context, image string) ([]*gql_generated.ImageSummary, error) {
+	layoutUtils := common.NewBaseOciLayoutUtils(r.storeController, r.log)
+	imageList := make([]*gql_generated.ImageSummary, 0)
+
+	repoList, err := layoutUtils.GetRepositories()
+	if err != nil {
+		r.log.Error().Err(err).Msg("unable to get repositories list")
+
+		return nil, err
+	}
+
+	if len(repoList) == 0 {
+		r.log.Info().Msg("no repositories found")
+
+		return imageList, nil
+	}
+
+	imageDir, imageTag := common.GetImageDirAndTag(image)
+
+	imageManifest, err := layoutUtils.GetImageManifest(imageDir, imageTag)
+	if err != nil {
+		r.log.Info().Str("image", image).Msg("image not found")
+
+		return imageList, err
+	}
+
+	imageLayers := imageManifest.Layers
+
+	for _, repo := range repoList {
+		repoInfo, err := r.ExpandedRepoInfo(ctx, repo)
+		if err != nil {
+			r.log.Error().Err(err).Msg("unable to get image list")
+
+			return nil, err
+		}
+
+		manifests := repoInfo.Images
+
+		var addImageToList bool
+		// verify every image
+		for _, manifest := range manifests {
+			addImageToList = true
+			layers := manifest.Layers
+			imageSize := 0
+
+			if imageTag != *manifest.Tag || imageDir != repo {
+				for _, l := range layers {
+					foundLayer := false
+
+					for _, k := range imageLayers {
+						if *l.Digest == k.Digest.Encoded() {
+							foundLayer = true
+							layerSize, _ := strconv.Atoi(*l.Size)
+							imageSize += layerSize
+
+							break
+						}
+					}
+
+					if !foundLayer {
+						addImageToList = false
+
+						break
+					}
+				}
+			}
+
+			if addImageToList {
+				name := repo
+				manifest.RepoName = &name
```

Comment:
This field should already be populated from ExpendedRepoInfo.
Same goes for computing the image size at line 570.